sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -61,7 +61,13 @@ public static CommunicationIdentifier FromRawId(string rawId)
 
             var segments = rawId.Split(':');
             if (segments.Length < 3)
+            {
+                if (segments.Length == 2 && segments[0] == "28")
+                {
+                    return new MicrosoftBotIdentifier(segments[1], true, CommunicationCloudEnvironment.Public);
```

Comment:
this section is starting to get a little messy...5 hardcoded numeric constants on 5 rows. we should keep an eye on this section in the future and possibly annotate it better, refactor it, or use some almighty regex to reduce the number of conditions.

Reply:
Let's keep this in mind for the future!